code/datums/components/storage/storage.dm

```diff
@@ -550,6 +566,8 @@
 //This proc return 1 if the item can be picked up and 0 if it can't.
 //Set the stop_messages to stop it from printing messages
 /datum/component/storage/proc/can_be_inserted(obj/item/I, stop_messages = FALSE, mob/M)
+	can_hold_full = typecacheof(can_hold)
+	cant_hold_full = typecacheof(cant_hold)

```

Comment:
Is there a reason you need both? You can just overwrite can_hold/cant_hold with the new value

Also I'd really like it if these get cached so we don't have multiple new lists for every single storage component

Reply:
There may be scenarios where the individual items given to the can_hold/cant_hold variables may want to be referenced, without having all the of typecaching been done to it. For example, for my examine code, I want to see the individual items listed in the can_hold variable so I can tell the user what items can be placed inside. If I overwrite it with the full typecache I'd be given ~100 items because of all their variants etc.

Reply:
A fair enough reason but if you want to do that you really need to cache these lists